developers/blobstream-rollups.md

```diff
@@ -0,0 +1,123 @@
+---
+description: Learn how to build rollups that use Blobstream
+---
+
+## Optimistic rollups
+
+One type of rollups that can be built with Blobstream is optimistic rollups.
+An optimistic rollup is a rollup that commits optimistically to a set of blocks,
+and allows the other parties to verify that the blocks are valid,
+and if they're not, they can create fraud proofs to signal that.
+
+Through the use of Blobstream, Celestia allows optimistic rollups to use Celestia as a DA layer,
+i.e., posting all the rollup data to Celestia, and only send commitments to the settlement contract.
+Then,
+users can use Blobstream as a source of truth to verify the rollup data
+and create fraud proofs in case of a misbehavior.
+
+To build an optimistic rollup that uses Celestia as a DA layer, the following constructions can be inspired by.
+
+### Optimistic rollups that uses a sequence of spans
+
+One way to construct an optimistic rollup that uses Celestia as a DA layer is to post the rollup data in Celestia.
+Then in the rollup header, use the following information to reference a sequence of spans, aka a data pointer:
+
+- `Height`: the height of the Celestia block containing the rollup data
+- `Start share`: the index of the first share containing the rollup data
+- `Data length`: the number of shares that the rollup data occupies
+
+Note:
+If the rollup data is submitted in multiple blocks,
+the above sequence of spans can be generalized to include multiple blocks.
+For simplicity, we will stick with the data only submitted to a single Celestia block.
+
+Now that this information is saved in the rollup header/settlement contract,
+and users/rollup full nodes will be able to query for the rollup data from Celestia and verify that it's valid.
+And if it's not, fraud proofs can be generated.
+
+The proofs in this setup are discussed in the [inclusion proofs](https://github.com/celestiaorg/blobstream-contracts/blob/master/docs/inclusion-proofs.md) documentation.
+
+### Optimistic rollups that use share commitments
+
+Another way to build a rollup is to replace the sequence of spans with a `height` and a `share commitment`.
```

Comment:
when talking about spans or sequences in general, I think we should mention their different functionalities without actually talking about rollups first, then we tie in other sections how spans can be used for different rollup types